unit/solvers/refinement/string_constraint_instantiation/instantiate_not_contains.cpp

```diff
@@ -81,6 +78,17 @@ std::set<exprt> full_index_set(const string_exprt &s)
   return ret;
 }
 
+/// Create the cartesian product of two sets.
+template<class T, class U>
+std::set<std::pair<T, U>> product(const std::set<T> ts, const std::set<U> us)
```

Comment:
depending on the usage, you could replace this `set` by a `vector` since all insertions are going to be distinct.

Reply:
The sets are used because sets are used in `string_refinementt`, where a set is needed because insertions aren't unique. Hence I had to use sets here, and why this utility function uses sets.